### PR TITLE
Use Switchblade on tests

### PR DESCRIPTION
**Pull request by brayanhenao**
Refactor the test framework from `cutlass` to `switchblade` to enable support to run the tests using Docker locally.

In addition, existing tests that might be redundant were refactored and simplified.

### CODE REVIEW

File: src/python/integration/init_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	proxyName, err := switchblade.RandomName()
+	Expect(err).NotTo(HaveOccurred())
+
+	proxyDeploymentProcess := platform.Deploy.WithBuildpacks("go_buildpack")
+
+	// TODO: remove this once go-buildpack runs on cflinuxfs4
+	// This is done to have the proxy app written in go up and running
+	if settings.Stack == "cflinuxfs4" {
+		proxyDeploymentProcess = proxyDeploymentProcess.WithStack("cflinuxfs3")
+	}
```
Comment (ryanmoran) [inline src/python/integration/init_test.go:76]: Do we need this conditional anymore? I think the TODO comment is indicating that we can remove it.

File: src/python/integration/init_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	dynatraceName, err := switchblade.RandomName()
+	Expect(err).NotTo(HaveOccurred())
+
+	dynatraceDeploymentProcess := platform.Deploy.WithBuildpacks("go_buildpack")
+
+	// TODO: remove this once go-buildpack runs on cflinuxfs4
+	// This is done to have the dynatrace broker app app written in go up and running
+	if settings.Stack == "cflinuxfs4" {
+		dynatraceDeploymentProcess = dynatraceDeploymentProcess.WithStack("cflinuxfs3")
+	}
```
Comment (ryanmoran) [inline src/python/integration/init_test.go:91]: Same as other TODO item. Do we still need this?

File: src/python/integration/pip_test.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+						CreateRequirementsTxtFile(Expect, filepath.Join(source, "sub_folder"), "other_requirement1.txt", "Markupsafe", "-r other_requirement2.txt")
+						CreateRequirementsTxtFile(Expect, filepath.Join(source, "sub_folder"), "other_requirement2.txt", "Werkzeug")
+						CreateRequirementsTxtFile(Expect, source, "requirements.txt", "Flask", "Jinja2", "gunicorn", "itsdangerous", "pylibmc", "cffi", "-r sub_folder/other_requirement1.txt")
+					})
+
+					it.After(func() {
+						Expect(os.RemoveAll(source)).To(Succeed())
+					})
+
+					it("installs the regulations-core package", func() {
```
Comment (ryanmoran) [inline src/python/integration/pip_test.go:91]:
```suggestion
					it("handles recursive requirements successfully", func() {
```

File: src/python/integration/pip_test.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+	for _, module := range modules {
+		_, err = file.WriteString(module + "\n")
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	err = file.Close()
+
+	Expect(err).NotTo(HaveOccurred())
+}
```
Comment (ryanmoran) [inline src/python/integration/pip_test.go:226]: I might suggest moving this to `init_test.go` since it is used across more than just this test file.

**Review by ryanmoran — COMMENTED** [review]
This looks great! The tests are so much easier to understand and I'm really happy to see that we've gotten rid of a bunch of extraneous fixtures. Nice work!

**Comment by ryanmoran** [comment]
> @ryanmoran Not sure what the `config.json` integration matrix should look like. I used the one from [Nodejs](https://github.com/cloudfoundry/nodejs-buildpack/blob/master/config.json#L9).
> 
> Also, why do we have `windows` in oses? Do we want to keep it?

I think the `config.json` settings you copied from the Node.js buildpack repo look fine to me. I'm not sure why that file previously mentioned `windows` in the OS list. That seems like a mistake.

File: src/python/integration/init_test.go (hunk 1 above)
Comment (brayanhenao) [reply src/python/integration/init_test.go:76]: The toolsmith environments we use do not support cflinuxfs4 natively (something we should discuss with them) so we set it up manually.

When this configuration is finished, the environment now supports cflinuxfs4 but has NO Buildpack associated with the stack, so the only Buildpack that would be available at the time of running the tests would be the one they are running on, there would be no other Buildpacks such as the Go one.

File: src/python/integration/init_test.go (hunk 2 above)
Comment (brayanhenao) [reply src/python/integration/init_test.go:91]: Same as above

File: src/python/integration/pip_test.go (hunk 3 above)
Comment (brayanhenao) [reply src/python/integration/pip_test.go:91]: Modified, thanks!

File: src/python/integration/pip_test.go (hunk 4 above)
Comment (brayanhenao) [reply src/python/integration/pip_test.go:226]: Moved, thanks!

**Comment by robdimsdale** [comment]
The tests are now failing with `pip 23.1`:

```
            -----> Installing python 3.10.11
                   Download [https://buildpacks.cloudfoundry.org/dependencies/python/python_3.10.11_linux_x64_cflinuxfs3_bfbfb6bf.tgz]
            -----> Installing pip 23.1
                   Download [https://buildpacks.cloudfoundry.org/dependencies/pip/pip_23.1_linux_noarch_any-stack_6b16c735.tgz]

...

                                     Looking in links: /tmp/pip
                                     ERROR: Could not find a version that satisfies the requirement flit_core<4,>=3.8 (from versions: none)
                                     ERROR: No matching distribution found for flit_core<4,>=3.8
```

**Pull request merged**